contrib/views/pg_system_views_extractor.sql

```diff
@@ -0,0 +1,116 @@
+-- Copyright (C) 2025 The pgexporter community
+
+-- Redistribution and use in source and binary forms, with or without modification,
+-- are permitted provided that the following conditions are met:
+
+-- 1. Redistributions of source code must retain the above copyright notice, this list
+-- of conditions and the following disclaimer.
+
+-- 2. Redistributions in binary form must reproduce the above copyright notice, this
+-- list of conditions and the following disclaimer in the documentation and/or other
+-- materials provided with the distribution.
+
+-- 3. Neither the name of the copyright holder nor the names of its contributors may
+-- be used to endorse or promote products derived from this software without specific
+-- prior written permission.
+
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+-- EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+-- OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+-- THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+-- SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+-- OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+-- HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+-- TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+-- This script extracts definitions of core system views used by pgexporter
+
+\echo '-- PostgreSQL Version Information'
+SELECT version();
+\echo ''
+
+-- formatting to check diff
+\pset format unaligned
+\pset tuples_only off
+\pset footer off
+
+-- temporary table with all pg_* relations
+CREATE TEMPORARY TABLE temp_result AS
+SELECT relname
+FROM pg_class
+WHERE relname LIKE 'pg\_%'
+ORDER BY relname;
+
+-- views to exclude from analysis
+CREATE TEMPORARY TABLE excluded_views AS
+SELECT unnest(ARRAY[
+    'pg_indexes',
+    'pg_locks',
+    'pg_matviews',
+    'pg_prepared_statements',
+    'pg_prepared_xacts',
+    'pg_rules',
+    'pg_settings',
+    'pg_shadow',
+    'pg_stats',
+    'pg_tables',
+    'pg_user',
+    'pg_views'
+]) AS viewname;
+
+-- target views by excluding unwanted views, relations will be dropped
+CREATE TEMPORARY TABLE target_views AS
+SELECT a.viewname
+FROM temp_result a
+LEFT JOIN excluded_views e ON a.viewname = e.viewname
+WHERE e.viewname IS NULL;
```

Comment:
Multiple tables do make it an effort to follow through, and so, readability suffers.

Instead of all these tables, I am of the opinion something like this would be easier to read:

```sql
CREATE TEMPORARY TABLE temp_result AS
SELECT relname
FROM pg_class
WHERE relkind='v'
  AND relname LIKE 'pg\_%'
  AND relname NOT IN (
    'pg_indexes', 
    'pg_locks', 
    'pg_matviews', 
    'pg_prepared_statements', 
    'pg_prepared_xacts', 
    'pg_rules', 
    'pg_settings', 
    'pg_shadow', 
    'pg_stats', 
    'pg_tables', 
    'pg_user', 
    'pg_views'
  )
ORDER BY relname;
```

I believe the result remains the same.

Reply:
Also what I've added assumes Jesper is fine with `relkind` of course.